### PR TITLE
DnsException improvements #2493

### PR DESCRIPTION
**Pull request by KowalczykBartek**
With this PR I wanted to provide fix for https://github.com/eclipse/vert.x/issues/2493

### CODE REVIEW

**Comment by vietj** [comment]
what's the motivation ?

**Comment by KowalczykBartek** [comment]
Oops, sorry :D updated.

**Comment by vietj** [comment]
right, I forgot it :-), thanks for the quickwin

**Comment by vietj** [comment]
for the stacktrace I think we want to modify the exception such as it does not actually capture a stack trace, the easy way to achieve is to extend rather `NoStackTraceThrowable` class

**Comment by KowalczykBartek** [comment]
ok, true, setting empty as stacktrace was a little bit nasty :D I updated code according to your suggestion. One thing, it is now not instance of Exception, is it ok ?

File: src/main/java/io/vertx/core/dns/DnsException.java [hunk 1]
```diff
@@ -19,11 +20,13 @@
...
  * @author <a href="mailto:nmaurer@redhat.com">Norman Maurer</a>
  */
-public final class DnsException extends Exception {
+public final class DnsException extends NoStackTraceThrowable {
 
+  private static final String ERROR_MESSAGE_TEMPLATE = "DNS query error occurred: %s";
   private DnsResponseCode code;
 
   public DnsException(DnsResponseCode code) {
+    super(String.format(ERROR_MESSAGE_TEMPLATE, code));
```
Comment (vietj) [inline src/main/java/io/vertx/core/dns/DnsException.java]: just use string concat instead of format

**Pull request merged**